skyportal/handlers/api/source.py

```diff
@@ -763,4 +770,26 @@ def get(self, obj_id):
             'Cache-Control', 'no-store, no-cache, must-revalidate, max-age=0'
         )
 
-        return self.write(image)
+        # From
+        # https://bhch.github.io/posts/2017/12/serving-large-files-with-tornado-safely-without-blocking/
+        chunk_size = 1024 * 1024 * 1  # 1 MiB
+        while True:
```

Comment:
Seems like we should have a timeout check. Like if it takes more than X seconds (X ~ 60) then throw a notification and break. As written it's not clear if this loop cannot go forever ...

Reply:
The loop terminates in one of two ways:

1. No more bytes left in buffer
2. Stream is closed

We always try to write inside the loop, so if that fails we're out.  Other exceptions will bubble through, so I can keep an eye on the logs to see when that happens.

Reply:
This is also a stopgap measure, until I can get the caching infrastructure fixed up—then we can serve up directly via nginx.

Reply:
OK, I've now made this a for-loop, predicated on the max file size we want to be able to send.